index.d.ts

```diff
@@ -906,6 +906,12 @@ declare namespace R {
         forEach<T>(fn: (x: T) => void, list: List<T>): T[];
         forEach<T>(fn: (x: T) => void): (list: List<T>) => T[];
         // forEach<T>: CurriedFunction2<(x: T) => void, List<T>, T[]>;
+  
+        /**
+         * Iterate over an input object, calling a provided function fn for each key and value in the object.
+         */
+        forEachObjIndexed<A>(fn: (val: A, key: string|number) => void, o: Obj<A>): Obj<A>;
+        forEachObjIndexed<A>(fn: (val: A, key: string|number) => void): (o: Obj<A>) => Obj<A>;
```

Comment:
If we're adding `number` to support the case of mapping arrays to objects, use `Prop` over `string|number`?